pkg/trace/api/api.go

```diff
@@ -390,6 +402,38 @@ func (r *HTTPReceiver) rateLimited(n int64) bool {
 	return !r.RateLimiter.Permits(n)
 }
 
+// StatsProcessor implementations are able to process incoming client stats.
+type StatsProcessor interface {
+	// ProcessStats takes a stats payload and consumes it. It is considered to be originating
+	// from the given lang.
+	ProcessStats(p pb.ClientStatsPayload, lang string)
+}
+
+// handleStats handles incoming stats payloads.
+func (r *HTTPReceiver) handleStats(w http.ResponseWriter, req *http.Request) {
+	defer timing.Since("datadog.trace_agent.receiver.stats_process_ms", time.Now())
+
+	ts := r.tagStats(v05, req)
+	rd := NewLimitedReader(req.Body, r.conf.MaxRequestBytes)
+	metrics.Count("datadog.trace_agent.receiver.stats_payload", 1, ts.AsTags(), 1)
+	if v := req.Header.Get("Content-Length"); v != "" {
```

Comment:
Same thing, handleTraces uses `req.Body.(*LimitedReader).Count` to get the length of the payload, I think it would make sense to use the same approach in both functions.

Reply:
Good point. Done.